lib/travis/model/job/queue.rb

```diff
@@ -30,6 +32,25 @@ def queues
       def default
         @default ||= new(Travis.config.default_queue)
       end
+
+      def sudo_detected?(config)
+        config.values_at(*custom_stages).compact.map do |value|
+          Array(value).reject { |s| s =~ /\s*#.*/ }.map { |s| s =~ /\b(sudo|ping)\b/ }.any?
+        end.any?
+      end
+
+      def custom_stages
+        @custom_stages ||= %w(
+          before_install
+          install
+          before_script
+          script
+          before_cache
+          after_success
+          after_failure
+          after_script
+        ).map(&:to_sym)
+      end
```

Comment:
I couldn't think of a reasonable solution.  I'm sure it will get out of sync.  I'd prefer to have this than to share dependencies between travis-build and anything else.